### ISSUE TITLE
Add option to automatically disable carousel when no sliding is possible.

### ISSUE THREAD

**Issue by gonssal** [metafizzy/flickity#556]
Hi, thank you for probably the best carousel implementantion right now. This will be a bit long so sorry about that.

I needed to remove the carousel when there's no possible sliding, that is, when all the cells or cellGroups are already visible. Default behaviour (disabled arrows) is not enough for two reasons:

- I could use `display: none` on disabled arrows and they wouldn't show up, but then I want actually disabled arrows (when you reach left/right slider limits) to be slightly visible changing their opacity and not hiding them.
- When you try to drag the slider, it still performs a dragging animation even though there's no cells to slide to.

So I initially tried using the watchCSS option:

```javascript
$el.flickity({
        contain: true,
        pageDots: false,
        groupCells: true,
        watchCSS: true
});
$el.data('flickity').on('resize', function() {
        if ($el.data('flickity').slides.length <= 1) {
                $el.addClass('disabled')
        }
        else {
                $el.removeClass('disabled')
        }
});
if ($el.data('flickity').slides.length <= 1) {
        $el.addClass('disabled')
}
```

The problem is that when Flickity doesn't find `content: 'flickity'`, it disables the carousel, so both the `on('resize')` and the `slides.length` functionality stops working. This means that when a carousel is disabled, if you then resize the window to a size where the carousel should be enabled again, it will never happen. This might be related to #514 and possibly other resizing issues.

So then I tried disabling/enabling manually and got to this:

```javascript
$el.flickity({
        contain: true,
        pageDots: false,
        groupCells: true
});
$el.data('flickity').on('resize', function() {
        if ($el.data('flickity').slides.length <= 1) {
                $el.flickity('deactivate');
                $el.data('flickity').isActive = true;
        }
        else {
                $el.flickity('select', 0);
                $el.data('flickity').isActive = false;
                $el.flickity('activate');
        }
});
if ($el.data('flickity').slides.length <= 1) {
        $el.flickity('deactivate');
        $el.data('flickity').isActive = true;
}
```

It is working, but when resizing the window I get some non-fatal javascript errors like this one: `Uncaught DOMException: Failed to execute 'appendChild' on 'Node': The new child element contains the parent.` I also needed to set the selected element to the first one because if the slider has been moved when there's a lot of cells/groupCells, when you resize and there are less cells/groupCells than you previously slided, the carousel will stop working.

And I also had to manually toggle the `isActive` property because, after reading the code, I realized that the `proto.resize` function has this guard clause:

```javascript
if ( !this.isActive ) {
    return;
}
```
...which basically means that, when you disable a carousel, the `resize` function will never again do anything, which I believe is also why the `watchCSS` option doesn't work properly when the window is resized.

So to sum it up, I think the guard clause in the `proto.resize` function should be removed and let it trigger even when a carousel is "not active", so it can be properly activated/deactivated when resizing.

I would (and will if you want me to) create a pull request, but considering how a small a change it is and that I'm not sure if it could affect anything else in the code, I decided against it.

Also I'll try to see if I can fix or get to the reason why the aforementioned errors pop up, but I don't have time for it right now, sorry.

**Comment by desandro** [comment]
Thank you for your kind word and for reporting this issue. Sounds like you're asking for a feature we already have an issue for: #187

**Comment by gonssal** [comment]
Would you consider merging a pull request removing the `if ( !this.isActive )`  from the `proto.resize` function for now?

**Comment by desandro** [comment]
I think the best way to go about this is to prevent Flickity from being initialized if there's not enough content. I'm spitballing here, but the code would look something like

``` js
var $carousel = $('.carousel');
// calculate total width of all cells
var cellsWidth = 0;
$carousel.find('.carousel-cell').each( function( i, cell ) {
  cellsWidth += $( cell ).outerWidth();
});
// init Flickity if enough cells
if ( cellsWidth >= $carousel.innerWidth() ) {
  $carousel.flickity({
    // ...
  });
}
```

**Comment by gonssal** [comment]
But then, if I'm not mistaken, you will have the exact same problem I'm exposing in the first post, where if you 'disable' Flickity by listening to the instance's `resize` event, then the event stops firing and your listener stops working, hence not being able to 'reenable' Flickity when it resizes to somewhere you want it re-enabled. That's why in my example code I have to manually change the `isActive` property so the event fires.

This is a pretty common case for handheld devices that can change orientation (smartphones, tablets...). For example I might want Flickity enabled on portrait mode and it disabled on landscape mode, and keep enabling/disabling it when the user changes orientation.

**Comment by desandro** [comment]
Sorry, Flickity does not support this feature. Add your 👍  to #278
